python/pyspark/ml/clustering.py

```diff
@@ -291,6 +292,317 @@ def _create_model(self, java_model):
+    k = Param(Params._dummy(), "k", "number of clusters to create")
+    optimizer = Param(Params._dummy(), "optimizer",
+                      "LDA optimizer, Currently 'em', 'online' are supported.")
+    learningOffSet = Param(Params._dummy(), "learningOffSet",
+                           "learning parameter that downweights early iterations")
+    learningDecay = Param(Params._dummy(), "learningDecay", "Learning rate")
+    subsamplingRate = Param(Params._dummy(), "subsamplingRate",
+                            "Fraction of the corpus to be sampled")
+    optimizeDocConcentration = Param(Params._dummy(), "optimizeDocConcentration",
+                                     "Indicates whether the docConcentration \
+                                     will be optimized during training")
+
+    @keyword_only
+    def __init__(self, featuresCol="features", k=10,
+                 optimizer="online", learningOffset=1024.0, learningDecay=0.51,
+                 subsamplingRate=0.05, optimizeDocConcentration=True,
+                 checkpointInterval=10, maxIter=20, seed=None):
+        """
+        :param featureCol:          feature column name in the dataframe
+        :param k:                   Number of clusters you want
+        :param optimizer:           LDAOptimizer used to perform the actual calculation.
+            Currently "em", "online" are supported. Default to "em".
+
+        :param learningOffset:      A (positive) learning parameter that downweights early
+            iterations. Larger values make early iterations count less. This is called "tau0"
+            in the Online LDA paper (Hoffman et al., 2010) Default: 1024.0, following Hoffman et al.
+        :param learningDecay:       Learning rate, set as an exponential decay rate. This should
+            be between (0.5, 1.0] to guarantee asymptotic convergence. This is called "kappa" in
+            the Online LDA paper (Hoffman et al., 2010). Default: 0.51, based on Hoffman et al.
+        :param subsamplingRate:     Fraction of the corpus to be sampled and used in each iteration
+            of mini-batch gradient descent, in range (0, 1]. Note that this should be adjusted in
+            synch with [[LDA.maxIter]] so the entire corpus is used.  Specifically, set both so that
+            maxIterations * miniBatchFraction >= 1. Default:0.05
+        :param optimizeDocConcentration:       Indicates whether the docConcentration (Dirichlet
+            parameter for document-topic distribution) will be optimized during training. Setting
+            this to true will make the model more expressive and fit the training data better.
+            Default: false
+        :param checkpointInterval:  Param for set checkpoint interval (>= 1) or disable
+            checkpoint (-1). E.g. 10 means that the cache will get checkpointed every 10 iterations.
+        :param maxIter:             Number of iterations. Default to 20
+        :param seed:                Param for random seed
```

Comment:
The problem is some parameters are not declared as Param (no get/set method for them). I am not sure the best practise for them. Seems other algorithms are the same.